diff --git data/JPL_ephemeris/Makefile data/JPL_ephemeris/Makefile
index 604ab2e..d6ec4ea 100644
--- data/JPL_ephemeris/Makefile
+++ data/JPL_ephemeris/Makefile
@@ -32,7 +32,8 @@ include ../../Makefile.include
 # TYPE OF EPHEMERIS [405|406|430|431]
 EPH_TYPE ?= 430
 # FTP client
-FTP_CLIENT = wget
+#FTP_CLIENT = wget -nv -c
+FTP_CLIENT = curl -L -O -\# -C - -f
 # Output name of the program:
 EPH_BIN = de$(EPH_TYPE).dat
 # Name of ascii-to-binary converter program source file:
@@ -101,7 +102,7 @@ $(TESTER): modules $(TEST_SRC)
 
 # Build input file for ascii-to-binary converter:
 $(EPH_INPUT):
-	$(FTP_CLIENT) ftp://ssd.jpl.nasa.gov/pub/eph/planets/ascii/de$(EPH_TYPE)/*
+	for F in $(EPH_ASCII_DE$(EPH_TYPE)) testpo.$(EPH_TYPE); do $(FTP_CLIENT) ftp://ssd.jpl.nasa.gov/pub/eph/planets/ascii/de$(EPH_TYPE)/$$F; done
 	cat $(EPH_ASCII_DE$(EPH_TYPE)) > $(EPH_INPUT)
 
 # Write back-up:
diff --git ups/eupspkg.cfg.sh ups/eupspkg.cfg.sh
index d36de94..0346b40 100644
--- ups/eupspkg.cfg.sh
+++ ups/eupspkg.cfg.sh
@@ -8,7 +8,6 @@ _ensure_exists()
 prep()
 {
 	# check for system prerequisites
-	_ensure_exists wget
 	_ensure_exists gfortran
 
 	default_prep
