diff --git ups/eupspkg.cfg.sh ups/eupspkg.cfg.sh
index 4de3f87..07ebdfb 100644
--- ups/eupspkg.cfg.sh
+++ ups/eupspkg.cfg.sh
@@ -1,14 +1,31 @@
-# EupsPkg config file. Sourced by 'eupspkg'
+#!/usr/bin/env bash
 
 config()
 {
-    # Disable external SSL on OS X as it is not a standard library
-    # and LSST does not need full SSL support.
-    ssloptions=""
-    if [[ $(uname) == Darwin ]]; then
-        ssloptions="-DWITH_SSL=bundled"
-    fi
 
+    ARGS=()
+
+    ARGS+=("-DCMAKE_INSTALL_PREFIX=${PREFIX}")
     # TokuDB requires third-party software jemalloci (see http://www.canonware.com/jemalloc/)
-    cmake . -DCMAKE_INSTALL_PREFIX=${PREFIX} -DPLUGIN_TOKUDB=NO $ssloptions
+    ARGS+=('-DPLUGIN_TOKUDB=NO')
+
+    case $(uname) in
+        Linux*)
+            # mroonga does not build on EL5 (conda ref platform) and is unused by LSST
+            ARGS+=('-DPLUGIN_MROONGA=NO')
+            # required to build on EL5
+            ARGS+=('-DHAVE_CXX_NEW=NO')
+            ;;
+        Darwin*)
+            # Disable external SSL on OS X as it is not a standard library
+            # and LSST does not need full SSL support.
+            ARGS+=('-DWITH_SSL=bundled')
+            ;;
+        *)
+            # non-fatal
+            echo "unsupported platform: $(uname)"
+            ;;
+    esac
+
+    cmake . "${ARGS[@]}"
 }
-- 
2.5.0

