--- python/lsst/meas/astrom/multiindex.py	2015-10-11 18:11:18.112874539 -0500
+++ python/lsst/meas/astrom/multiindex.py	2015-10-11 18:11:03.155919495 -0500
@@ -144,7 +144,7 @@
         @param coord   Coordinate to check (lsst.afw.coord.Coord)
         @param distance   Angular distance (lsst.afw.geom.Angle)
         """
-        return (self._nside == 0 or healpixDistance(self._healpix, self._nside, coord) <= distance)
+        return (self._healpix == -1 or healpixDistance(self._healpix, self._nside, coord) <= distance)

     def __getitem__(self, i):
         self.reload()
